dotnet/src/Agents/OpenAI/OpenAIResponseAgent.cs

```diff
@@ -144,5 +144,50 @@ private async IAsyncEnumerable<ChatMessageContent> InternalInvokeAsync(
             yield return message;
         }
     }
+
+    // TODO Share this implementation with CoreKernelArgumentsExtensions 
+    private static readonly Dictionary<string, PromptExecutionSettings> s_emptySettings = [];
+
+    /// <summary>
+    /// Provides a merged instance of <see cref="KernelArguments"/> with precedence for override arguments.
+    /// </summary>
+    /// <param name="overrideArguments">The override arguments.</param>
+    /// <remarks>
+    /// This merge preserves original <see cref="PromptExecutionSettings"/> and <see cref="KernelArguments"/> parameters.
+    /// It allows for incremental addition or replacement of specific parameters while also preserving the ability
+    /// to override the execution settings.
+    /// </remarks>
+    private KernelArguments MergeArguments(KernelArguments? overrideArguments)
+    {
+        // Avoid merge when override arguments are not set.
+        if (overrideArguments is null)
+        {
+            return this.Arguments ?? [];
+        }
+
+        // Avoid merge when the Agent arguments are not set.
+        if (this.Arguments is null)
+        {
+            return overrideArguments ?? [];
```

Comment:
Same here, suggest making this KernelArguments? (nullable).

So if the original is null as well as the provided overriden, just return null.

```suggestion
            return overrideArguments ?? [];
```